src/advancedKubectl/content.md

```diff
@@ -0,0 +1,1108 @@
+
+Let's consider an example.
+
+Imagine you want to create a [ReplicaSet](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#replicaset-v1-apps) resource. To do so, you would define the ReplicaSet in a file named `replicaset.yaml` file, and then run the following command:
+
+~~~bash
+kubectl create -f replicaset.yaml
+~~~
+
+Obviously, this creates your ReplicaSet in Kubernetes. But what happens behind the scenes?
+
+Kubernetes has a *create ReplicaSet* operation, and like all Kubernetes operations, it is exposed as an API endpoint. The specific API endpoint for this operation is as follows:
+
+~~~
+POST /apis/apps/v1/namespaces/{namespace}/replicasets
+~~~
+
+> You can find the API endpoints of all Kubernetes operations in the [API reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13) (including the [above endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#create-replicaset-v1-apps)). To make an actual request to an endpoint, you need to prepend the URL of the API server to the endpoint paths that are listed in the API reference.
+
+Consequently, when you execute the above command, kubectl makes an HTTP POST request to the above API endpoint. The ReplicaSet definition (that you provided in the `replicaset.yaml` file) is passed in the body of the request.
+
+This is how kubectl works for *all* commands that interact with the Kubernetes cluster. In all these cases, kubectl simply makes HTTP requests to the appropriate Kubernetes API endpoints.
+
+> Note that it's totally possible to control Kubernetes with a tool like `curl` by manually issuing HTTP requests to the Kubernetes API. Kubectl just makes it easier for you to use the Kubernetes API.
+
+These are the basics of what kubectl is and how it works. But there is much more about the Kubernetes API that every kubectl user should know. To this end, let's briefly dive into the Kubernetes internals.
+
+
+### Kubernetes internals
+
+Kubernetes consists of a set of independent components that run as separate processes on the nodes of a cluster. Some components run on the master nodes and others run on the worker nodes, and each component has a very specific function.
+
+The most important components on the master nodes are:
+
+- **Storage backend:** stores resource definitions (usually [etcd](https://coreos.com/etcd/) is used)
+- **API server:** provides Kubernetes API and manages storage backend
+- **Controller manager:** ensures resource statuses match specifications
+- **Scheduler:** schedules Pods to worker nodes
+
+And the most important component running on each worker node is:
+
+- **kubelet:** manages execution of containers on a worker node
```

Comment:
You could condense the bullet point into a single sentence. If you decide to keep like this, it's missing an article

**the** kubelet

Reply:
I changed the spelling and wording accordingly.